core/domain/classifier_services.py

```diff
@@ -213,12 +306,6 @@ def create_classifier(job_id, classifier_data):
         classifier.state_name, classifier.algorithm_id,
         classifier.classifier_data, classifier.data_schema_version)
 
-    # Create the mapping from <exp_id,exp_version,state_name>
-    # to classifier_id.
-    create_classifier_exploration_mapping(
-        classifier.exp_id, classifier.exp_version_when_created,
-        classifier.state_name, classifier_id)
-
     return classifier_id
```

Comment:
That's a very big change and affects many things. It seems plausible to me. We had classifier data in a separate model because we thought that we will have same classifier data for multiple (version, state). But looking it closely we also have same job for all these (version, state), the job which stored classifier data for the first time. Now we are mapping states to jobs directly I don't see any need to keep classifier data separate. Since different states may have same job and there is one to one correspondence between job and classifier. So we can migrate the classifier data to training job.

Also would it be better to do that in a separate PR, I guess? This PR will become hard to review if we handle all stuff in same PR.